panther_battery/test/test_battery.cpp

```diff
@@ -85,174 +71,54 @@ void TestBattery::CheckBatteryStateMsg(
-  EXPECT_EQ(BatteryStateMsg::POWER_SUPPLY_HEALTH_OVERVOLTAGE, battery_state_.power_supply_health);
-  EXPECT_TRUE(battery_->HasErrorMsg());
-}
-
-TEST_F(TestBattery, BatteryMsgHealthOverheat)
-{
-  UpdateBattery(1.5, 0.01, 0.98, 0.5, false);
-
-  EXPECT_EQ(BatteryStateMsg::POWER_SUPPLY_HEALTH_OVERHEAT, battery_state_.power_supply_health);
-  EXPECT_TRUE(battery_->HasErrorMsg());
-}
-
-TEST_F(TestBattery, BatteryMsgHealthCold)
-{
-  UpdateBattery(1.5, 0.01, 2.81, 0.5, false);
+  battery_->SetErrorMsg("error");
 
-  EXPECT_EQ(BatteryStateMsg::POWER_SUPPLY_HEALTH_COLD, battery_state_.power_supply_health);
-  EXPECT_TRUE(battery_->HasErrorMsg());
-}
-
-TEST_F(TestBattery, BatteryMsgStatusFull)
-{
-  UpdateBattery(1.66, 0.01, 0.98, 0.5, true);
-
-  EXPECT_FLOAT_EQ(1.0, battery_state_.percentage);
-  EXPECT_EQ(BatteryStateMsg::POWER_SUPPLY_STATUS_FULL, battery_state_.power_supply_status);
-}
-
-TEST_F(TestBattery, BatteryMsgStatusCharging)
-{
-  UpdateBattery(1.5, 0.01, 0.98, 0.5, true);
-
-  EXPECT_EQ(BatteryStateMsg::POWER_SUPPLY_STATUS_CHARGING, battery_state_.power_supply_status);
+  ASSERT_TRUE(battery_->HasErrorMsg());
+  EXPECT_NE("", battery_->GetErrorMsg());
 }
 
-TEST_F(TestBattery, BatteryMsgStatusNotCharging)
+TEST_F(TestBattery, GetBatteryPrcent)
 {
-  UpdateBattery(1.5, 0.01, 0.98, 0.04, true);
-
-  EXPECT_EQ(BatteryStateMsg::POWER_SUPPLY_STATUS_NOT_CHARGING, battery_state_.power_supply_status);
+  EXPECT_FLOAT_EQ(0.0, battery_->GetBatteryPercent(30.0f));
+  EXPECT_FLOAT_EQ(0.0, battery_->GetBatteryPercent(32.0f));
+  EXPECT_FLOAT_EQ(0.638297872, battery_->GetBatteryPercent(38.0f));
+  EXPECT_FLOAT_EQ(1.0, battery_->GetBatteryPercent(41.4f));
+  EXPECT_FLOAT_EQ(1.0, battery_->GetBatteryPercent(45.0f));
 }
 
-TEST_F(TestBattery, TestGetErrorMsg)
+TEST_F(TestBattery, ResetBatteryMsgs)
 {
-  EXPECT_FALSE(battery_->HasErrorMsg());
-  EXPECT_EQ("", battery_->GetErrorMsg());
+  // expect empty message at the beginning
+  EXPECT_EQ(BatteryStateMsg(), battery_->GetBatteryMsg());
+  EXPECT_EQ(BatteryStateMsg(), battery_->GetBatteryMsgRaw());
 
-  // send overvoltage
   auto stamp = rclcpp::Time(0);
-  battery_voltage_raw_ = 1.72;
-  battery_current_raw_ = 0.01;
-  battery_temp_raw_ = 1.5;
-  battery_charge_raw_ = 0.5;
-
-  battery_->Update(stamp, false);
+  battery_->ResetBatteryMsgs(stamp);
   battery_state_ = battery_->GetBatteryMsg();
+  CheckBatteryStateMsg(
+    BatteryStateMsg::POWER_SUPPLY_STATUS_UNKNOWN, BatteryStateMsg::POWER_SUPPLY_HEALTH_UNKNOWN);
 
-  ASSERT_TRUE(battery_->HasErrorMsg());
-  EXPECT_NE("", battery_->GetErrorMsg());
+  battery_state_ = battery_->GetBatteryMsgRaw();
+  CheckBatteryStateMsg(
+    BatteryStateMsg::POWER_SUPPLY_STATUS_UNKNOWN, BatteryStateMsg::POWER_SUPPLY_HEALTH_UNKNOWN);
 }
 
 int main(int argc, char ** argv)
 {
-  rclcpp::init(argc, argv);
   testing::InitGoogleTest(&argc, argv);
 
   auto run_tests = RUN_ALL_TESTS();
 
-  rclcpp::shutdown();
   return run_tests;
```

Comment:
```suggestion
  return RUN_ALL_TESTS();
```
and remove previous line